patroni/api.py

```diff
@@ -89,6 +89,9 @@ def do_GET(self, write_status_code_only=False):
                 status_code = 503
             elif response['role'] in path:  # response['role'] != 'master'
                 status_code = 503 if patroni.noloadbalance else 200
+            elif path.startswith('/sync') and cluster.is_synchronous_mode() and response.get('role') == 'replica' \
```

Comment:
I'm not sure this is the right place to plug it:

/sync will return 200 if running on the master with a lock, 503 if it is a master without the lock.
/sync/replica will return 503 even on a sync replica if noloadbalance tag is set for it (which is completely reasonable), or 200 if noloadbalance is not set, even if it's not running in a synchronous mode.

Reply:
> /sync will return 200 if running on the master with a lock, 503 if it is a master without the lock.

It won't, we have a line 84 for that.

> or 200 if noloadbalance is not set, even if it's not running in a synchronous mode.

Why? It check explicitly that `cluster.sync and cluster.sync.sync_standby == patroni.postgresql.name`

Reply:
> It won't, we have a line 84 for that.

How is so? the user calling /sync will get to the line 84 and if it is a master with a lock will return 200. Otherwise (a master without the lock) it will get to line 89 and return 503.  Both have probably nothing to do with what he asked.

> Why? It check explicitly that cluster.sync and cluster.sync.sync_standby == patroni.postgresql.name

Because of line 90 and 91, that I think will be triggered for the `/sync/replica` since replica is in path.